x-pack/platform/packages/shared/kbn-elastic-assistant/impl/assistant/settings/ai_for_soc_settings_management.tsx

```diff
@@ -0,0 +1,154 @@
+import { ManagementSettingsTabs } from './types';
+
+interface Props {
+  dataViews: DataViewsContract;
+  onTabChange?: (tabId: string) => void;
+  currentTab: ManagementSettingsTabs;
+}
+
+/**
+ * Modal for overall Assistant Settings, including conversation settings, quick prompts, system prompts,
+ * anonymization, knowledge base, and evaluation via the `isModelEvaluationEnabled` feature flag.
+ */
+export const AIForSOCSettingsManagement: React.FC<Props> = React.memo(
+  ({ dataViews, onTabChange, currentTab }) => {
+    const { http, selectedSettingsTab, setSelectedSettingsTab } = useAssistantContext();
+
+    useEffect(() => {
+      if (selectedSettingsTab) {
+        // selectedSettingsTab can be selected from Conversations > System Prompts > Add System Prompt
+        onTabChange?.(selectedSettingsTab);
+        // set to null after nav change as to not persist the tab
+        setSelectedSettingsTab(null);
+      }
+    }, [onTabChange, selectedSettingsTab, setSelectedSettingsTab]);
+
+    const { data: connectors } = useLoadConnectors({
+      http,
+    });
+    const defaultConnector = useMemo(() => getDefaultConnector(connectors), [connectors]);
+
+    const { euiTheme } = useEuiTheme();
+
+    const tabsConfig = useMemo(
+      () => [
+        {
+          id: CONVERSATIONS_TAB,
+          label: i18n.CONVERSATIONS_MENU_ITEM,
+        },
+        {
+          id: CONNECTORS_TAB,
+          label: i18n.CONNECTORS_MENU_ITEM,
+        },
+        {
+          id: SYSTEM_PROMPTS_TAB,
+          label: i18n.SYSTEM_PROMPTS_MENU_ITEM,
+        },
+        {
+          id: QUICK_PROMPTS_TAB,
+          label: i18n.QUICK_PROMPTS_MENU_ITEM,
+        },
+        {
+          id: ANONYMIZATION_TAB,
+          label: i18n.ANONYMIZATION_MENU_ITEM,
+        },
+        {
+          id: KNOWLEDGE_BASE_TAB,
+          label: i18n.KNOWLEDGE_BASE_MENU_ITEM,
+        },
+      ],
+      []
+    );
+
+    const tabs = useMemo(() => {
+      return tabsConfig.map((t) => ({
+        ...t,
+        onClick: () => {
+          onTabChange?.(t.id);
+        },
+        isSelected: t.id === currentTab,
+      }));
+    }, [onTabChange, currentTab, tabsConfig]);
+    return (
+      <EuiFlexGroup
+        data-test-subj="AIForSOCSettingsManagement"
+        css={css`
+          margin-top: ${euiTheme.size.l};
+        `}
+      >
+        <EuiFlexItem grow={false} css={{ width: '200px' }}>
+          <EuiListGroup flush>
+            {tabs.map(({ id, label, onClick, isSelected }) => (
+              <EuiListGroupItem
+                key={id}
+                label={label}
+                onClick={onClick}
+                data-test-subj={`settingsPageTab-${id}`}
+                isActive={isSelected}
+                size="s"
+              />
+            ))}
+          </EuiListGroup>
+        </EuiFlexItem>
+        <EuiFlexItem data-test-subj={`tab-${currentTab}`}>
+          {currentTab === CONNECTORS_TAB && <AIForSOCConnectorSettingsManagement />}
```

Comment:
I wonder if this wouldn't be cleared if we use a switch instead of short-circuit rendering. I assume there is no chance that we render more than 1 tab content at a time? What do you think?

Reply:
done!